### PR TITLE
optimize : optimize logs when global  commit/rollback catch an exception

### PR DESCRIPTION
**Pull request by Bughue**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
TC在 RM的commit遇到异常和返回false时的行为不一致，容易发生误解，这个pr只对日志进行优化

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/seata/seata/issues/4879

### CODE REVIEW

File: server/src/main/java/io/seata/server/coordinator/DefaultCore.java [hunk 1]
```diff
@@ -328,7 +329,7 @@ public boolean doGlobalRollback(GlobalSession globalSession, boolean retrying) t
                     }
                 } catch (Exception ex) {
                     StackTraceLogger.error(LOGGER, ex,
-                        "Rollback branch transaction exception, xid = {} branchId = {} exception = {}",
+                        "Rollback branch transaction exception, xid = {} branchId = {} exception = {}, global rollback failed",
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/coordinator/DefaultCore.java:332]: Why is it not consistent with the log above ？

File: server/src/main/java/io/seata/server/coordinator/DefaultCore.java [hunk 2]
```diff
@@ -232,8 +232,9 @@ public boolean doGlobalCommit(GlobalSession globalSession, boolean retrying) thr
                             }
                     }
                 } catch (Exception ex) {
-                    StackTraceLogger.error(LOGGER, ex, "Committing branch transaction exception: {}",
-                        new String[] {branchSession.toString()});
+                    String commitInfo = retrying ? "Global commit continue" : "Global commit failed";
+                    StackTraceLogger.error(LOGGER, ex, "Committing branch transaction exception: {},{}",
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/coordinator/DefaultCore.java:236]: Are placeholders missing?

Comment (Bughue) [reply server/src/main/java/io/seata/server/coordinator/DefaultCore.java:236]: commitInfo是实际填充的内容，placeholder本来是一个，现在改成两个，已经测过

File: server/src/main/java/io/seata/server/coordinator/DefaultCore.java (hunk 1 above)
Comment (Bughue) [reply server/src/main/java/io/seata/server/coordinator/DefaultCore.java:332]: 这就是一开始令人费解的地方，因为：doGlobalCommit和doGlobalRollback遇到异常的时候行为不是一致的

- doGlobalCommit （retrying=false）单个rm异常，会打断commit，等待global的retry
- doGlobalCommit （retrying=true）单个rm异常，**会继续执行下一个rm的commit**
- doGlobalRollback （retrying=false）单个rm异常，会打断rollback，等待global的retry
- doGlobalRollback （retrying=true）单个rm异常，会打断rollback，等待global的retry

这也是我提交这个pr的原因，因为从之前的日志看不出来是continue了还是中断了

Comment (slievrly) [reply server/src/main/java/io/seata/server/coordinator/DefaultCore.java:340]: ok, I got.

Comment (slievrly) [reply server/src/main/java/io/seata/server/coordinator/DefaultCore.java:340]: If possible, I hope to print the retry information in the rollback exception to facilitate problem locating.

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**